src/validation.rs

```diff
@@ -159,7 +159,7 @@ pub fn validate(claims: &Map<String, Value>, options: &Validation) -> Result<()>
                 }
                 Value::Array(_) => {
                     let provided_aud: HashSet<String> = from_value(aud.clone())?;
-                    if provided_aud.intersection(correct_aud).count() == 0 {
+                    if provided_aud.intersection(correct_aud).next().is_none() {
```

Comment:
Does that make any difference in practice when the size of each set will be <2-3? I would go for the clearer approach with count if that's not the case